pandas/core/strings.py

```diff
@@ -323,15 +328,23 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     n : int, default -1 (all)
         Number of replacements to make from start
-    case : boolean, default True
-        If True, case sensitive
-    flags : int, default 0 (no flags)
-        re module flags, e.g. re.IGNORECASE
+    case : boolean, optional
+        - if False, case insensitive
+        - Must be None if `pat` is a compiled regex
```

Comment:
@jreback would it make sense to revert back to `case=True` here and test for `case is True` when `pat` is a compiled regex? That would keep it consistent across `strings.py`. 

However, I foresee this PR as a potential stepping stone to allow compiled regex in all search-pattern functions in `strings.py`, and if it's a better pattern for `case=None` (over `case=True`) then let's start making that change with this function.